.github/workflows/sbom.yml

```diff
@@ -45,39 +45,43 @@ jobs:
         run: |
-          syft \
-            -o spdx-json \
-            --file audit-scanner-sbom-${{ matrix.arch }}.spdx \
-            ghcr.io/${{ github.repository_owner }}/audit-scanner@${{ env.PLATFORM_DIGEST }}
+          set -e
+          DIGEST=$(crane manifest ghcr.io/${{github.repository_owner}}/audit-scanner@${{ inputs.image-digest }} \
+            | jq '.manifests[] | select(.annotations["vnd.docker.reference.type"]=="attestation-manifest") | select(.annotations["vnd.docker.reference.digest"]=="${{ env.PLATFORM_DIGEST }}") | .digest'
+          )
+          echo "ATTESTATION_MANIFEST_DIGEST=${DIGEST}" >> "$GITHUB_ENV"
 
-      - name: Sign SBOM file
+      - name: Sign attestation manifest
         run: |
-          cosign sign-blob --yes \
-            --output-certificate audit-scanner-sbom-${{ matrix.arch }}.spdx.cert \
-            --output-signature audit-scanner-sbom-${{ matrix.arch }}.spdx.sig \
-            audit-scanner-sbom-${{ matrix.arch }}.spdx
+          cosign sign --yes \
+            --output-certificate audit-scanner-attestation-${{ matrix.arch }}.cert \
+            --output-signature audit-scanner-attestation-${{ matrix.arch }}.sig \
+          ghcr.io/${{github.repository_owner}}/audit-scanner@${{ env.ATTESTATION_MANIFEST_DIGEST}}
 
-      - name: Attach SBOM file in the container image
-        shell: bash
+      - name: Find provenance file digest
         run: |
-          cosign attach \
-            sbom --sbom audit-scanner-sbom-${{ matrix.arch }}.spdx \
-            ghcr.io/${{ github.repository_owner }}/audit-scanner@${{ env.PLATFORM_DIGEST }}
+          set -e
+          DIGEST=$(crane manifest ghcr.io/${{github.repository_owner}}/audit-scanner@${{ env.ATTESTATION_MANIFEST_DIGEST}} | \
+            jq '.layers[] | select(.annotations["in-toto.io/predicate-type"] == "https://slsa.dev/provenance/v0.2") | .digest')
+          echo "PROVENANCE_DIGEST=${DIGEST}" >> "$GITHUB_ENV"
 
-      - name: Sign SBOM file pushed to OCI registry
-        shell: bash
+      - name: Find sbom file digest
         run: |
           set -e
-          SBOM_TAG="$(echo ${{ env.PLATFORM_DIGEST }} | sed -e 's/:/-/g').sbom"
+          DIGEST=$(crane manifest ghcr.io/${{github.repository_owner}}/audit-scanner@${{ env.ATTESTATION_MANIFEST_DIGEST}} |  \
+            jq '.layers[] | select(.annotations["in-toto.io/predicate-type"] == "https://spdx.dev/Document") | .digest')
+          echo "SBOM_DIGEST=${DIGEST}" >> "$GITHUB_ENV"
 
-          cosign sign --yes \
-            ghcr.io/${{github.repository_owner}}/audit-scanner:${SBOM_TAG}
+      - name: Download attestation and SBOM files
+        run: |
+          set -e
+          crane blob ghcr.io/${{github.repository_owner}}/audit-scanner@${{ env.PROVENANCE_DIGEST}} > audit-scanner-attestation-${{ matrix.arch }}-provenance.json
+          crane blob ghcr.io/${{github.repository_owner}}/audit-scanner@${{ env.SBOM_DIGEST}} > audit-scanner-attestation-${{ matrix.arch }}-sbom.json
```

Comment:
Let's also sign the actual `json` files again, so that we can upload these signatures into the GH release.

The signatures we produced earlier are not for the actual files, but for the OCI manifest that references them.

Reply:
> Let's also sign the actual json files again, so that we can upload these signatures into the GH release.
Yes, working on it already

> The signatures we produced earlier are not for the actual files, but for the OCI manifest that references them.
Yes, that's why I've decided to talk about the signatures during our daily. That's come to my mind after working ours. xD

Reply:
Sure, I just wanted to leave a written track of what we discussed during our call :)